### DISCUSSION TITLE
mrv2 v1.2.5

### DISCUSSION THREAD

**Discussion by ggarra13** [ggarra13/mrv2#294] [Announcements]
v1.2.5
======

- RE-RELEASE: v1.2.5 in English had a serious bug in OCIO selection through the menus.
- SECOND RE-RELEASE: Added toggle between showing OCIO or COLOR information topbar in OCIO menu.
- THIRD RE-RELEASE: Fixed macOS Intel (amd64) distribution looking for mesa OpenGL
libraries.
- Input Color Space and Look setting are now stored with each clip.  However, 
  you cannot compare two clips (say in a wipe) with different display settings.
- Made Image/Previous and Image/Next not appear when a single image or movie
  was loaded.
- Fixed typo when saving session files which would prevent them from being
  loaded which had gotten broken in v1.2.3.
- OCIO configuration can now be selected from the menus.
- OCIO Display/View can now be applied on a monitor per monitor basis.
- Removed OCIO from the Color Panel.
- Removed OCIO from the Top bar.  This change hopefully won't be controversial.
  Now OCIO is all managed through the OCIO menu which can be easily accessed
  with the Right Mouse Button or from the Main menu bar.
- Major refactoring of convoluted OCIO code.
- Allowed Chaging OCIO Input Color Space from the OCIO menu.
- Fixed OCIO ICS when set to None and changing languages. 
- Fixed annotation markers in timeline showing even when switching clips.
- Fixed Wayland PNG incompatibility on older Linux OSes.  Tested to work on
  Rocky Linux 8.10, Ubuntu 22.04.4 LTS and Ubuntu 24.04.4 LTS.
- Fixed saving annotations leading to black or stopping playback when there
  were no annotations on Windows.
- Added session pattern to Open Movie or Sequence.
- Moved View/OCIO Presets to OCIO/Presets.
- Added OCIO/Current File/Look to new OCIO menu.
- Added -otio or -edl command-line option to automatically create an .otio 
  timeline from a list of clips (movies or sequences) provided in the 
  command-line.  Note that FPS is taken to be that of the one with highest 
  FPS, so sequences may leave gaps if video clips bigger than their FPS are
  used.
- Made -otio or -edl command-line option work with relative paths.
- Fixed .otio, .otioz and -edl flags when concatenation videos with different
  rotations.
- Changed name of "Reproducción" to "Reproducir" in Spanish translation.
- Added Chinese (Simplified) Translation done with AI.
- Added German Translation done with AI.
- Added Hindi Translation done with AI.
- Made Languages display their English name in parenthesis to help if switching
  to an unknown language by mistake.

<hr /><em>This discussion was created from the release <a href='https://github.com/ggarra13/mrv2/releases/tag/v1.2.5'>mrv2 v1.2.5</a>.</em>

**Comment by AltaArts** [comment]
First, just want to say that MRV2 keeps improving more and more!  It is fantastic that you are putting so much thought into this, and it shows.

I do agree with having a dedicated OCIO menu is a good idea, and do appreciate you adding the option to display it back on the top bar.  Most of the time in normal work we use the OCIO presets menu via a shortcut, so we wouldn't have the topbar visible anyway.  But for example yesterday I was working on our custom OCIO config and it made it just slightly harder going into the menus.  But now we have the option that has been solved (though if I really wanted to picky, I would love it if when the option is enabled it would put the topbar back as it was in prior versions since all of our displays have the screenspace for it).

The one thing I would suggest is for the LUT UI display handling.  In the previous versions the OCIO and LUT were all together in the Color Panel, but now that OCIO has moved then LUT is sort of stranded there still.  I know a LUT is technically nothing to do with OCIO, but sometimes a LUT is used in conjunction with a OCIO setting.  For example if we have a show look LUT, we tend to save an OCIO preset that includes the LUT (which sometimes is quicker than adding it to the OCIO config itself).  So I would think maybe display the active LUT file in the OCIO UI?

And I do see a small UI issue with the OCIO menu.  The current OCIO settings appear to be in the "inactive" UI state and thus greyed out making it hard to read (see screenshots).  It is not a big deal and doesn't change the functionality, but I would think they should be the same style as "active" elements.  I first thought it was the "oxy" theme that I am using, but it is the same with the other themes as well.


![MRV2 - OCIO Top Menu](https://github.com/user-attachments/assets/aeab2f35-61c4-4dfe-8591-19c3712abd7b)
![MRV2 - OCIO rcl Menu](https://github.com/user-attachments/assets/8e390031-2ad0-4bad-86f9-6fa5d6ee4d6f)

But again, I do want to thank you for MRV2!
